rotkehlchen/chain/ethereum/adex/utils.py

```diff
@@ -0,0 +1,130 @@
+TOM_POOL_FEE_REWARDS_API_URL = 'https://tom.adex.network/fee-rewards'
+SECONDS_PER_YEAR = FVal('31536000')
+PERIOD_END_AT_FORMAT = '%Y-%m-%dT%H:%M:%S.%fZ'
+
+ADEX_EVENTS_PREFIX = 'adex_events'
+
+
+def deserialize_adex_event_from_db(
+        event_tuple: AdexEventDBTuple,
+) -> Union[Bond, Unbond, UnbondRequest]:
+    """Turns a tuple read from DB into an appropriate AdEx event.
+    May raise a DeserializationError if something is wrong with the DB data.
+
+    Event_tuple index - Schema columns
+    ----------------------------------
+    0 - tx_hash
+    1 - address
+    2 - identity_address
+    3 - timestamp
+    4 - bond_id
+    5 - type
+    6 - pool_id
+    7 - amount
+    8 - nonce
+    9 - slashed_at
+    10 - unlock_at
+    """
+    db_event_type = event_tuple[5]
+    if db_event_type not in {str(event_type) for event_type in EventType}:
+        raise DeserializationError(
+            f'Failed to deserialize event type. Unknown event: {db_event_type}.',
+        )
+
+    tx_hash = HexStr(event_tuple[0])
+    address = deserialize_ethereum_address(event_tuple[1])
+    identity_address = deserialize_ethereum_address(event_tuple[2])
+    timestamp = deserialize_timestamp(event_tuple[3])
+    bond_id = HexStr(event_tuple[4])
+
+    if db_event_type == str(EventType.BOND):
+        try:
+            assert event_tuple[6] is not None
+            assert event_tuple[7] is not None
+            assert event_tuple[8] is not None
+            assert event_tuple[9] is not None
+        except AssertionError as e:
+            raise DeserializationError(
+                f'Failed to deserialize bond event. Unexpected data: {event_tuple}.',
+            ) from e
+
+        return Bond(
+            tx_hash=tx_hash,
+            address=address,
+            identity_address=identity_address,
+            timestamp=timestamp,
+            bond_id=bond_id,
+            pool_id=HexStr(event_tuple[6]),
+            amount=deserialize_asset_amount(event_tuple[7]),
+            nonce=event_tuple[8],
+            slashed_at=Timestamp(event_tuple[9]),
+        )
+    if db_event_type == str(EventType.UNBOND):
+        return Unbond(
+            tx_hash=tx_hash,
+            address=address,
+            identity_address=identity_address,
+            timestamp=timestamp,
+            bond_id=bond_id,
+        )
+    if db_event_type == str(EventType.UNBOND_REQUEST):
+        try:
+            assert event_tuple[10] is not None
+        except AssertionError as e:
+            raise DeserializationError(
+                f'Failed to deserialize bond event. Unexpected data: {event_tuple}.',
+            ) from e
+
+        return UnbondRequest(
+            tx_hash=tx_hash,
+            address=address,
+            identity_address=identity_address,
+            timestamp=timestamp,
+            bond_id=bond_id,
+            unlock_at=Timestamp(event_tuple[10]),
+        )
+
+    raise AssertionError(f'Unexpected event type case: {db_event_type}.')
```

Comment:
Here raise a deserialization error. It should indeed not happen, but it's not an assertion error. It can be caused by VERY unusual but, still external factors. For example a user opening their DB by hand and editing random fields.